### DISCUSSION TITLE
GLFW Throws platform unsupported on linux

### DISCUSSION THREAD

**Discussion by atomsk-0** [dotnet/Silk.NET#1226] [Q&A] [answered]
# Summary
When i'm trying to create new window i get this Error:
System.PlatformNotSupportedException: GLFW is not supported on this platform.
   at Silk.NET.Windowing.Glfw.GlfwPlatform.ThrowUnsupported()
   at Silk.NET.Windowing.Glfw.GlfwPlatform.CreateWindow(WindowOptions options)
   at Silk.NET.Windowing.Window.Create(WindowOptions options)

i have called Registerplatform for GlfwWindowing and GlfwInput before creating new window

# Steps to reproduce
- Platform: Desktop - Fedora 37 Workstation (Wayland Windowing System)
- Framework Version: .NET 7.0
- API: OpenGL
- API Version: OpenGL 4.6
1.Call platform registers for GlfwWindow and GlfwInput
2.Create new window
3.Run Window

Not sure if  this is a bug or Silk.NET just simply don't support glfw windowing on linux with wayland windowing system.

**Comment by Beyley** [comment]
You need to install GLFW using your system package manager, which seems to be fedora, so you can use `yum`, and install glfw with `yum glfw`

**Comment by atomsk-0** [comment]
installed and still same issue

**Comment by Perksey** [comment, marked as answer]
I have converted this to a discussion as I'm 100% positive this can work, but I'm not too familiar with Linux package managers. It is possible that the version of GLFW that it's trying doesn't support Wayland or something. Perhaps run with `LD_DEBUG=all` and see if the output says anything?

**Reply by atomsk-0** [reply]
You're right seems like the version silk.net is trying to load doesn't support Wayland. I have already builded a native lib of glfw with Wayland support but i don't know how i can make Silk.NET to use it.

**Reply by Perksey** [reply]
Try copying it to the executable directory with file name `libglfw.so.3.3`. I believe the order it tries it in is:
1. First, tries libglfw.so.3.3 with the plain LD_PATH (this should get a hit in your local directory implicitly)
2. If it's not picked up, it looks in our native libraries (which are apparently built for X)
3. if it's still not picked up, then we look for libglfw.so.3 and repeat steps 1-2
4. And so on with libglfw.so

**Reply by atomsk-0** [reply]
Ye replacing the exsisting libglfw.so.3 with the wayland one fixed the issue thanks.
